### PR TITLE
feat(api): add GET endpoint to retrieve accounts by member address

### LINKED ISSUE #154: [BACKEND] Create API Endpoint to Get Member's Accounts
### **Title**
Create GET API endpoint to retrieve all accounts where a user is a member

#### **Overview**
Create a new API endpoint that allows retrieving all Spherre accounts where a given user (identified by their contract address) is a member. This endpoint will be essential for the frontend to get the accounts that a user is a member of and enable account switching.

#### **Technical Requirements**

**Endpoint Details:**
- **Method:** GET
- **Path:** `/api/v1/accounts/member/{member_address}`
- **Parameters:** 
  - `member_address` (path parameter): The contract address of the member
- **Response:** JSON array of account objects

**Expected Response Format:**
```json
[
    {
      "id": "uuid",
      "address": "0x1234...",
      "name": "Account Name",
      "description": "Account description",
      "is_private": true,
      "threshold": 2,
      "created_at": "2024-01-01T00:00:00Z",
      "updated_at": "2024-01-01T00:00:00Z"
    }
  ]
}
```

#### **Implementation Details**

**1. Service Layer Enhancement**
- The `AccountService.get_member_accounts()` method already exists and provides the core functionality
- No additional service methods needed

**2. API View Implementation**
- Create a new view file: `backend/spherre/app/views/accounts.py`
- Implement the endpoint using Flask-RESTful or standard Flask routes
- Add proper error handling for invalid member addresses

**3. Serialization**
- Create account serializer in `backend/spherre/app/serializers/account.py`
- Ensure consistent JSON response format
- Include all necessary account fields for frontend consumption

**4. Blueprint Registration**
- Update `backend/spherre/app/__init__.py` to register the accounts blueprint
- Add proper URL prefix: `/api/v1`

#### **Error Handling**
- **404 Not Found:** When member address doesn't exist
- **400 Bad Request:** When member address format is invalid
- **500 Internal Server Error:** For database or server errors

#### **Validation Requirements**
- Validate member address format (Starknet address validation)
- Ensure member exists in database before querying accounts
- Handle empty results gracefully (return empty array, not error)

#### **Testing Requirements**
- Unit tests for the new endpoint
- Test cases for:
  - Valid member address with accounts
  - Valid member address without accounts
  - Invalid member address format
  - Non-existent member address
  - Database connection errors

#### **Documentation**
- Add API documentation with example requests/responses
- Include the endpoint in the API schema
- Document expected error responses in the serializer class.

#### **Acceptance Criteria**
- [ ] Endpoint returns all accounts where the member is a participant
- [ ] Proper error handling for invalid/non-existent members
- [ ] Consistent JSON response format
- [ ] Unit tests pass
- [ ] Integration tests with frontend
- [ ] API documentation updated
- [ ] Performance acceptable for expected load

### PR DESCRIPTION
**Pull request by bernardev254**
## Description
This PR introduces a new API endpoint that retrieves all Spherre accounts in which a given user (identified by their contract address) is a member. This functionality enables the frontend to quickly access all relevant accounts for a user and supports account switching features.

## Related Issue

Fixes #154

## Type of Change

- [ ] Bug fix
- [X ] New feature
- [ ] Refactor
- [ ] Documentation update
- [ ] Other (please describe)

## How Has This Been Tested?

- [ X] Unit tests
- [ ] Integration tests
- [ ] Manual tests
- [ ] Other (please describe)

## Checklist:

- [X ] I have read the contributing guidelines.
- [ ] I have updated the documentation (if applicable).
- [ X] My changes do not break existing functionality.
- [ X] I have added tests that cover my changes (if applicable).
- [ X] All new and existing tests pass.
- [ ] I have included screenshots or GIFs to demonstrate the changes (if applicable).

## Summary by CodeRabbit

* **New Features**
  * Added an API endpoint to retrieve accounts associated with a specific member address.
  * Introduced validation for StarkNet addresses in API requests.
  * Implemented serialization for account data in API responses.

* **Bug Fixes**
  * Improved error handling for invalid addresses and server errors in account retrieval.

* **Tests**
  * Added comprehensive tests for the new account retrieval endpoint, including success, error, and edge cases.

* **Chores**
  * Specified Python version for the backend environment.

### CODE REVIEW

File: backend/spherre/tests/test_models/test_views/test_accounts_view.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        db.session.remove()
+        db.drop_all()
+        self.ctx.pop()
+
+    def test_get_member_accounts_valid(self):
+        with patch.object(
+            AccountService,
+            "get_member_accounts",
+            return_value=[{"id": "1", "address": self.member.address}],
+        ):
```
Comment (codebestia) [inline backend/spherre/tests/test_models/test_views/test_accounts_view.py:34]: Instead of mocking this, actually create the account
```python
AccountService.create_account(
    address="0x123",
    name="Test Account",
     threshold=1.0,
     members=["0x456", "0x789"],
     description="This is a test account",
 )
```

Then call the `get_member_accounts` function normally.

File: backend/spherre/app/views/accounts.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        serializer = AccountSerializer(many=True)
+        serialized_accounts = serializer.dump(accounts)
+
+        return jsonify(serialized_accounts), 200
+    except Exception:
+        return jsonify({"error": "Server error"}), 500
+
+
+def is_valid_starknet_address(address):
+    return bool(re.fullmatch(r"0x[a-fA-F0-9]{64}", address))
```
Comment (codebestia) [inline backend/spherre/app/views/accounts.py:31]: Add this in a utils file: `spherre/app/utils/validation.py`

**Review by codebestia — CHANGES_REQUESTED** [review]
Great Job @bernardev254 
Love the implementation!
Just some minor changes to be done and we are good to go
